orm/src/main/java/org/hibernate/search/hcore/impl/SearchFactoryReference.java

```diff
@@ -22,16 +20,16 @@
  */
 public final class SearchFactoryReference implements Service {
 
-	private volatile CompletableFuture<ExtendedSearchIntegrator> extendedIntegratorFuture;
+	private volatile ExtendedSearchIntegrator extendedIntegrator;
 
-	public void initialize(CompletableFuture<ExtendedSearchIntegrator> future) {
-		extendedIntegratorFuture = future;
+	public void initialize(ExtendedSearchIntegrator extendedIntegrator) {
+		this.extendedIntegrator = extendedIntegrator;
 	}
 
 	public ExtendedSearchIntegrator getSearchIntegrator() {
-		final CompletableFuture<ExtendedSearchIntegrator> value = this.extendedIntegratorFuture;
+		ExtendedSearchIntegrator value = this.extendedIntegrator;
 		if ( value != null ) {
-			return value.join();
+			return value;
```

Comment:
yes I think that's a reasonable tradeoff.

It's not my typical style to push for integrating a problem and "let the users report issues" but in this case I think I'm quite confident on taking the risk, as I believe the timing issue is really not a problem in practice as this won't be run in parallel.

The reasoning is that the variable will be read very often, and beyond a volatile read it's all getting expensive. We'll revisit this (of course) if someone managed to proof I'm wrong on the parallel timings..